src/string.cr

```diff
@@ -2177,6 +2193,21 @@ class String
     last_index
   end
 
+  # ditto
+  def rindex(search : Regex, offset = 0)
+    offset += size if offset < 0
+    return nil unless 0 <= offset <= size
+
+    matchlist = self[0, self.size - offset].scan(search)
```

Comment:
This can be done without storing all the matches in an array. Just use the block form and save the last match.